syncstorage/storage/spanner.py

```diff
@@ -358,13 +371,68 @@ def _find_items(self, session, user, collection, **params):
                 bind_types[ts_var] = param_types.TIMESTAMP
 
         # Generate the query
+        fields = params.get("fields")
+        offset_params = params
+
+        if self._bsolm_index_separate and fields is None:
+            # Split the get_items query (TODO: get_item_ids could
+            # reference solely the index) query in 2:
+            # #1: query BsoLastModified directly for bso ids
+            # #2: query bso w/ 'id IN UNNEST([<ids from #1>])'
+
+            bsolm_params = params.copy()
+            bsolm_params["fields"] = ["id"]
+            bsolm_query = FIND_ITEMS(bso, bsolm_params)
+            bsolm_query = str(bsolm_query.compile())
+            bsolm_query = bsolm_query.replace(
+                "FROM bso",
+                "FROM bso@{FORCE_INDEX=BsoLastModified}"
+            )
+            bsolm_query = bsolm_query.replace(":", "@")
+
+            result = session.transaction.execute_sql(
+                bsolm_query,
+                params=bind,
+                param_types=bind_types,
+            )
+            ids = [row[0] for row in list(result)]
+
+            # keep the orig. for encode_next_offset
+            offset_params = params.copy()
+
+            # Setup a 'id IN (:id_1)' bind param for #2 (replacing it
+            # below)
+            params["ids"] = [BOGUS_ID]
```

Comment:
What stops us from just putting the returned `ids` in `params["id"]`, and basically just pretending that the caller had requested those ids specifically?

Reply:
This is specifically to utilize spanner's `UNNEST` (what it's replaced with below) for query 2.

I do have a TODO for the FIND_ITEMS/sqlalchemy generation piece to do this for us. params["ids"] from the client and/or this case should *both* utilize UNNEST. Unfortunately coercing the sqlalchemy level to generate it for us is a bit involved, thus the current str replace.

Another quick solution for this is having FIND_ITEMS handle UNNEST by returning a str with the appropriate replacements (also not ideal, further straying from the original FIND_ITEMS code..) 🤷‍♂